autogpt_platform/backend/backend/server/v2/store/db.py

```diff
@@ -706,45 +706,35 @@ async def get_my_agents(
     logger.debug(f"Getting my agents for user {user_id}, page={page}")
 
     try:
-        agents_with_max_version = await prisma.models.AgentGraph.prisma().find_many(
-            where=prisma.types.AgentGraphWhereInput(
-                userId=user_id, StoreListing={"none": {"isDeleted": False}}
-            ),
-            order=[{"version": "desc"}],
-            distinct=["id"],
-            skip=(page - 1) * page_size,
-            take=page_size,
+        search_filter = prisma.types.LibraryAgentWhereInput(
+            userId=user_id,
+            Agent={"is": {"StoreListing": {"none": {"isDeleted": False}}}},
+            isArchived=False,
+            isDeleted=False,
         )
 
-        # store_listings = await prisma.models.StoreListing.prisma().find_many(
-        #     where=prisma.types.StoreListingWhereInput(
-        #         isDeleted=False,
-        #     ),
-        # )
-
-        total = len(
-            await prisma.models.AgentGraph.prisma().find_many(
-                where=prisma.types.AgentGraphWhereInput(
-                    userId=user_id, StoreListing={"none": {"isDeleted": False}}
-                ),
-                order=[{"version": "desc"}],
-                distinct=["id"],
-            )
+        library_agents = await prisma.models.LibraryAgent.prisma().find_many(
+            where=search_filter,
+            order=[{"agentVersion": "desc"}],
+            skip=(page - 1) * page_size,
+            take=page_size,
+            include={"Agent": True},
         )
```

Comment:
Doesn't this cause agents acquired from the marketplace to show up in the "Publish an agent" dialog?